code/game/gamemodes/gang/gang.dm

```diff
@@ -224,6 +225,7 @@ var/list/gang_colors_pool = list("red","orange","yellow","green","blue","purple"
 				gangster_mind.current.Paralyse(5)
 				gangster_mind.current.visible_message("<FONT size=3><B>[gangster_mind.current] looks like they've given up the life of crime!<B></font>")
 			gangster_mind.current << "<FONT size=3 color=red><B>You have been reformed! You are no longer a gangster!</B><BR>You try as hard as you can, but you can't seem to recall any of the identities of your former gangsters...</FONT>"
+			gangster_mind.memory = ""
```

Comment:
Probably, but I don't know that there's another way. (This is the same thing that cult does upon deconversion.)